### DISCUSSION TITLE
`isAdditionalFlowStep` documentation is misleading

### DISCUSSION THREAD

**Discussion by Marcono1234** [github/codeql#8616] [Q&A] [answered]
The documentation for `DataFlow::Configuration::isAdditionalFlowStep` for multiple languages currently says:

> Holds if the additional flow step from `node1` to `node2` must be taken into account in the analysis.

In my opinion the "must be taken into account" is rather misleading. To me that sounds like flow must take this step, while actually (if I understand it correctly) it adds an optional additional step the flow may take. Maybe something like the following would make this clearer:

> Holds if the flow step from `node1` to `node2` is considered an additional step the flow may take.

I have created this as discussion because this misunderstanding might be rather subjective, and possibly I am also interpreting the behavior of this predicate incorrectly.

**Comment by smowton** [comment]
How about `Holds if the analysis should assume that data may flow from node1 to node2 in addition to the normal dataflow steps.`

**Reply by Marcono1234** [reply]
That sounds good to me, it is also clearer than what I proposed. But of course you can also keep the current description if you think it fits well enough.

**Comment by smowton** [comment, marked as answer]
https://github.com/github/codeql/pull/8638

**Reply by smowton** [reply]
Merged
